src-tauri/src/main.rs

```diff
@@ -329,19 +329,18 @@ fn trigger_action(
                         whole_buffer.push(response.clone());
                         delta_buffer.push(response);
 
-                        let delta_output = {
-                            if delta_buffer.len() > 4 {
-                                let s = delta_buffer.clone().join("");
-                                delta_buffer.clear();
-                                s
-                            } else {
-                                "".to_string()
-                            }
-                        };
+                        // No need to clone delta_buffer as we're going to clear it anyway.
+                        let delta_output = delta_buffer.join("");
+                        delta_buffer.clear();
 
                         for step in trigger.next_steps.clone() {
                             if let Step::StreamTextToScreen = step {
-                                enigo.text(&delta_output).expect("Failed to type out text");
+                                // Split the output on new lines and send each part followed by an 'Enter' keypress.
+                                for step in trigger.next_steps.clone() {
+                                    if let Step::StreamTextToScreen = step {
+                                        enigo.text(&delta_output).expect("Failed to type out text");
+                                    }
+                                }
```

Comment:
looks like this may have been duplicated on accident

Reply:
fixed that.